### PR TITLE
adding migration path and modifying other docs

### PR DESCRIPTION
**Pull request by monikavar**
Changes:
- Updated docs with direct methods for version `2.8.2` and included migration changes from `2.5.x` to `2.8.2`.
- Added requests and responses for all the supported direct methods for `2.5.x` and `2.8.2`.

### CODE REVIEW

File: docs/modules/publisher-directmethods.md [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+- GetDiagnosticInfo_V1
+- AddOrUpdateEndpoints_V1
+
+To understand the terminologies and migration path from version 2.5.x of OPC Publisher to 2.8.2s, please refer to [this](publisher-dmmigration.md) migration document.
+
+
+
+Now let's dive into each direct method request and response payloads with examples.
+
+#### PublishNodes_V1**
```
Comment (marcschier) [inline docs/modules/publisher-directmethods.md]: Trailing stars?  On several headings below as well.  Also, please check using the markdown linter in VS Code, using a # and then ## heading might be better.

File: docs/modules/publisher-directmethods.md [hunk 2]
```diff
@@ -0,0 +1,325 @@
+[Home](../../readme.md)
+
+### Configuration via IoT Hub Direct methods ###
+
+OPC Publisher implements the following [IoT Hub Direct Methods](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-direct-methods) which can be called from an application (anywhere in the world) leveraging the [IoT Hub Device SDK](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-sdks).
+
+The following direct methods are inherited from 2.5.x in addition to some new ones:
+
+- PublishNodes_V1
```
Comment (marcschier) [inline docs/modules/publisher-directmethods.md:11]: Maybe mention somewhere that the _V1 is not required (like in 2.5)?   Also, I would split the 2.5 and extra methods in 2 sections so it is clearer.

File: docs/modules/publisher-dmmigration.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+
+
+### Direct Methods of version 2.5.x ###
+
+The following  table describes the direct methods which were available in OPC Publisher 2.5.x with request and response. The direct methods which are deprecated in version 2.8.2 are GetDiagnosticLog, GetDiagnosticStartupLog, ExitApplication and GetInfo. 
+
+| 2.5.x                            |                                                              |                                                              |                          |
+| -------------------------------- | ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------------ |
+| **MethodName**                   | **Request**                                                  | **Response**                                                 | **Is Deprecated in 2.8** |
```
Comment (marcschier) [inline docs/modules/publisher-dmmigration.md]: Is deprecated means it is there, but should not be used.  But it is not there, should say ("in 2.8.2")

File: docs/modules/publisher-dmmigration.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+| `UseSecurity`                    | no        | Boolean | false                     | Desired opc session security mode                                                            |
+| `OpcAuthenticationMode`          | no        | Enum    | Anonymous                 | Enum to specify the session authentication.<br>Options: Anonymous, UserName                  |
+| `UserName`                       | no        | String  | null                      | The username for the session authentication<br>Mandatory if OpcAuthentication mode is UserName|
+| `Password`                       | no        | String  | null                      | The password for the session authentication<br>Mandatory if OpcAuthentication mode is UserName|
+| `DataSetWriterGroup`             | no        | String  | EndpointUrl_#             | The writer group collecting datasets defined for a certain <br>endpoint uniquely identified by the above attributes. <br>This shall be used to identify the session opened into the <br>server. The default value consists of the EndpointUrl string, <br>followed by a deterministic hash composed of the <br>EndpointUrl, security and authentication attributes.|
```
Comment (marcschier) [inline docs/modules/publisher-dmmigration.md]: _#

File: docs/modules/publisher-dmmigration.md [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+| `DataSetWriterId`                | no        | String  | DataSetPublishingInterval | The unique identifier for a data set writer used to collect <br>opc nodes to be semantically grouped and published with <br>a same publishing interval. <br>When not specified a string representing the common <br>publishing interval of the nodes in the data set collection. <br>This the DataSetWriterId shall uniquely identify a data set <br>within a DataSetGroup. The unicity shall be determined <br>using the provided DataSetWriterId and the publishing <br>interval of the grouped OpcNodes.  An individual <br>subscription shall be created for each DataSetWriterId|
+| `DataSetPublishingInterval`      | no        | Integer | false                     | The publishing interval used for a grouped set of nodes <br>under a certain DataSetWriter. When defined it shall <br>override the OpcPublishingInterval value in the OpcNodes <br>if grouped underneath a DataSetWriter. |
+| `Tag`                            | no        | String  | empty                     | TODO                                                                                         |
```
Comment (marcschier) [inline docs/modules/publisher-dmmigration.md]: TODO

**Comment by karok2m** [comment]
@monikavar, can you please add some content in the description of the PR ?

File: docs/modules/publisher-directmethods.md [hunk 6]
```diff
@@ -0,0 +1,325 @@
...
+> ```json
+> {
+>   "status": 200,
+>   "payload": {}
+> }
+> ```
+
+
+
+#### UnpublishNodes_V1**
```
Comment (dacolgit) [inline docs/modules/publisher-directmethods.md]: Missing `**`. Should be `**UnpublishNodes_V1**`

File: docs/modules/publisher-directmethods.md [hunk 7]
```diff
@@ -0,0 +1,325 @@
...
+> ```json
+> {
+>   "status": 200,
+>   "payload": {}
+> }
+> ```
+
+
+
+#### UnpublishAllNodes_V1**
```
Comment (dacolgit) [inline docs/modules/publisher-directmethods.md]: Missing `**.` Should be `**UnpublishAllNodes_V1**`

File: docs/modules/publisher-directmethods.md [hunk 8]
```diff
@@ -0,0 +1,325 @@
...
+>      {
+>          "EndpointUrl": "opc.tcp://opcplc:50001/"
+>      }
+>   ] 
+> }
+> ```
+
+
+
+#### GetConfiguredNodesOnEndpoints_V1**
```
Comment (dacolgit) [inline docs/modules/publisher-directmethods.md]: Missing `**.` Should be `**GetConfiguredNodesOnEndpoints_V1**`

File: docs/modules/publisher-directmethods.md [hunk 9]
```diff
@@ -0,0 +1,325 @@
...
+>        "heartbeatInterval":0,
+>        "heartbeatIntervalTimespan":"00:00:00"
+>     }
+>   ] 
+> } 
+> ```
+
+
+
+#### GetDiagnosticInfo_V1**
```
Comment (dacolgit) [inline docs/modules/publisher-directmethods.md]: Missing `**.` Should be `**GetDiagnosticInfo_V1**`

**Review by dacolgit — DISMISSED** [review]

File: docs/modules/publisher.md [hunk 10]
```diff
@@ -143,383 +143,11 @@ There are several command line arguments that can be used to set global settings
...
-  ]
-  ```
-
-  *Response:* when successful -  Status 200 and an empty json (`{}`) as payload
-  
-  *Exceptions:* a response corresponding to an exception will be returned if:
-
-  - request payload contains deletion request for an endpoint (DataSet) that is not present in publisher configuration.
-  - request payload contains two or more entries for the same endpoint (DataSet).
+To learn how to migrate from version 2.5 (or below) to version 2.8 for the direct methods, please check this [migration path](publisher-dmmigration.md).
```
Comment (karok2m) [inline docs/modules/publisher.md]: Let's change this to specifically mention `2.8.2` version and not `2.8` in general.

File: docs/modules/publisher-directmethods.md [hunk 11]
```diff
@@ -0,0 +1,325 @@
+[Home](../../readme.md)
+
+### Configuration via IoT Hub Direct methods ###
```
Comment (karok2m) [inline docs/modules/publisher-directmethods.md]: The document should start with a top level heading with one `#`. Please remove two `#`s from all headings if we want to keep the same hierarchy.

File: docs/modules/publisher-directmethods.md (hunk 2 above)
Comment (karok2m) [reply docs/modules/publisher-directmethods.md:11]: I also agree with the idea of separating them to two lists, one for inherited ones and one for new ones.

File: docs/modules/publisher-directmethods.md [hunk 12]
```diff
@@ -0,0 +1,325 @@
...
+
+OPC Publisher implements the following [IoT Hub Direct Methods](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-direct-methods) which can be called from an application (anywhere in the world) leveraging the [IoT Hub Device SDK](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-sdks).
```
Comment (karok2m) [inline docs/modules/publisher-directmethods.md]: We should mention that we are talking specifically about version `2.8.2` here.

File: docs/modules/publisher-directmethods.md (hunk 1 above)
Comment (karok2m) [reply docs/modules/publisher-directmethods.md]: Please add two `*`s in front of `PublishNodes_V1`. In original document the formatting looked like this:

```
#### **PublishNodes_V1**
```

Please do the same change for all direct method name headings.

File: docs/modules/publisher-directmethods.md [hunk 13]
```diff
@@ -0,0 +1,325 @@
...
+
+To understand the terminologies and migration path from version 2.5.x of OPC Publisher to 2.8.2s, please refer to [this](publisher-dmmigration.md) migration document.
+
+
+
+Now let's dive into each direct method request and response payloads with examples.
+
+#### PublishNodes_V1**
+
+PublishNodes enables a client to add a set of nodes to be published for a specific DataSetWriter. When a DataSetWriter already exists, the nodes shall be incrementally added to the very same dataset. When it does not already exist, a new DataSetWriter shall be created with the initial set of nodes contained in the request.
```
Comment (karok2m) [inline docs/modules/publisher-directmethods.md]: We talk immediately about `DataSetWriter`. The term should be explained first.

File: docs/modules/publisher-directmethods.md [hunk 14]
```diff
@@ -0,0 +1,325 @@
...
+
+  *Request:* follows strictly the request payload schema as described, the OpcNodes attribute being mandatory.
```
Comment (karok2m) [inline docs/modules/publisher-directmethods.md]: Where is the described schema ? Let's add description of the JSON payload schema above before talking about specific direct methods similar to original document.

File: docs/modules/publisher-directmethods.md [hunk 15]
```diff
@@ -0,0 +1,325 @@
...
+*Note*: If all the nodes from a DataSet are to be unpublished, the DataSetWriter entity shall be completely removed from the configuration storage.
+
+  *Request:*  follows strictly the request payload schema, the OpcNodes attribute being mandatory.
+
+  *Response:* when successful - Status 200 and an empty json (`{}`) as Payload
+
+  *Exceptions:* a response corresponding to an exception will be returned if:
+
+- request payload contains an endpoint (DataSet) that is not present in publisher configuration
+- request payload contains a node that is not present in publisher configuration
```
Comment (karok2m) [inline docs/modules/publisher-directmethods.md]: Those lines are missing some indentation so that they don't break whole structure of `Request`/`Response`/`Exceptions`/`Example` block. Please add two (or maybe four, please check) white spaces in front of the `-` characters.

File: docs/modules/publisher-directmethods.md [hunk 16]
```diff
@@ -0,0 +1,325 @@
...
+> *Response:*
+>
+> ```json
+> {
+>   "status": 200,
+>   "payload": {}
+> }
+> ```
+
+
```
Comment (karok2m) [inline docs/modules/publisher-directmethods.md]: No need for so many blank lines. Markdown likes not more than one empty line in between blocks.

File: docs/modules/publisher-directmethods.md [hunk 17]
```diff
@@ -0,0 +1,325 @@
...
+  *Request:* follows strictly the request payload schema as described, the OpcNodes attribute being mandatory.
+
+  *Response:* when successful Status 200 and an empty json (`{}`) as payload
+
+  *Exceptions:*
+
+  *Example:*
+
+> *Method Name:* `PublishNodes_V1`
```
Comment (karok2m) [inline docs/modules/publisher-directmethods.md]: Example blocks should have the same level of indentation as ```*Example:*``` itself to render on the same level. Please add two spaces in front of all `>` characters in all example blocks.

File: docs/modules/publisher-dmmigration.md [hunk 18]
```diff
@@ -0,0 +1,143 @@
+[Home](../../readme.md)
+
+# Migration path
+
+The latest version 2.8.2 adds support for configuration via direct methods. OPC Publisher implements multiple [IoT Hub Direct Methods](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-direct-methods) which can be called from an application leveraging the [IoT Hub Device SDK](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-sdks). This document provides the migration path from 2.5.x to 2.8.2.
+
+The direct methods' request payload of version 2.8.2 is backwards compatible with the 2.5.x direct methods. The payload schema allows configuration of additional extensions introduced in the pn.json in the publisher 2.6.x and newer e.g. DataSetWriterGroup, DataSetWriterId, QueueSize per node, etc.
+
+For these set of methods, the encoding shall be JSON and no compression or payload chunking mechanism shall be applied in order to ensure the backwards compatibility with the 2.5.x version of OPC Publisher module.
```
Comment (karok2m) [inline docs/modules/publisher-dmmigration.md]: - Either `these sets` or `this set` please. `these` implies a plural noun.
- `the encoding shall be JSON and no compression or payload chunking mechanism shall be applied` - the wording with `shall` sounds like a feature specification for us, and not a documentation for a client. Let's change the wording to describe what is there and not how it should be implemented by the developer.

File: docs/modules/publisher-dmmigration.md [hunk 19]
```diff
@@ -0,0 +1,143 @@
...
+
+The latest version 2.8.2 adds support for configuration via direct methods. OPC Publisher implements multiple [IoT Hub Direct Methods](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-direct-methods) which can be called from an application leveraging the [IoT Hub Device SDK](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-sdks). This document provides the migration path from 2.5.x to 2.8.2.
+
+The direct methods' request payload of version 2.8.2 is backwards compatible with the 2.5.x direct methods. The payload schema allows configuration of additional extensions introduced in the pn.json in the publisher 2.6.x and newer e.g. DataSetWriterGroup, DataSetWriterId, QueueSize per node, etc.
+
+For these set of methods, the encoding shall be JSON and no compression or payload chunking mechanism shall be applied in order to ensure the backwards compatibility with the 2.5.x version of OPC Publisher module.  
+
+
+
+### Direct Methods of version 2.5.x ###
```
Comment (karok2m) [inline docs/modules/publisher-dmmigration.md]: Inconsistent heading. Previous heading had one `#`, so this one should have two of them and not immediately three.

File: docs/modules/publisher-dmmigration.md [hunk 20]
```diff
@@ -0,0 +1,143 @@
...
+| **ExitApplication**              | SecondsTillExit (optional)                                   | StatusCode, List<StatusResponse>                             | Yes                      |
+| **GetInfo**                      | -                                                            | GetInfoMethodResponseModel >>    VersionMajor  VersionMinor  VersionPatch  SemanticVersion  InformationalVersion OS  OSArchitecture  FrameworkDescription | Yes                      |
+
+A [sample configuration application](https://github.com/Azure-Samples/iot-edge-opc-publisher-nodeconfiguration) as well as an [sample application for reading diagnostic information](https://github.com/Azure-Samples/iot-edge-opc-publisher-diagnostics) are provided from OPC Publisher open-source, leveraging this interface.
+
+_Hint: The samples applications are not actively maintained and may be outdated._
+
+
+
+### Direct Methods of version 2.8.2 ###
```
Comment (karok2m) [inline docs/modules/publisher-dmmigration.md]: The heading should not have `#` characters in the end.

File: docs/modules/publisher-dmmigration.md [hunk 21]
```diff
@@ -0,0 +1,143 @@
...
+
+The following direct methods are inherited from version 2.5.x plus some additional ones in the latest version 2.8.2:
```
Comment (karok2m) [inline docs/modules/publisher-dmmigration.md]: Let's specify which are inherited and which are additional.

File: docs/modules/publisher-dmmigration.md (hunk 5 above)
Comment (monikavar) [reply docs/modules/publisher-dmmigration.md]: we will add this TODO later once it is implemented.

File: docs/modules/publisher-directmethods.md [hunk 22]
```diff
@@ -0,0 +1,328 @@
...
+- UnpublishAllNodes_V1
+- GetConfiguredEndpoints_V1
+- GetConfiguredNodesOnEndpoints_V1
+- GetDiagnosticInfo_V1
+
+New direct methods:
+
+-  AddOrUpdateEndpoints_V1
+
+Note: `_V1` suffix is not required.
```
Comment (cristipogacean) [inline docs/modules/publisher-directmethods.md]: I think _V1 is required in 2.8.2. can you validate and rephrase this? I think it was missing in 2.5.x

File: docs/modules/publisher-dmmigration.md [hunk 23]
```diff
@@ -0,0 +1,141 @@
...
+| **GetDiagnosticStartupLog**      | -                                                            | MissedMessageCount, LogMessageCount, List<Log>               | No                      |
+| **ExitApplication**              | SecondsTillExit (optional)                                   | StatusCode, List<StatusResponse>                             | No                      |
+| **GetInfo**                      | -                                                            | GetInfoMethodResponseModel >>    VersionMajor  VersionMinor  VersionPatch  SemanticVersion  InformationalVersion OS  OSArchitecture  FrameworkDescription | No                      |
+
+A [sample configuration application](https://github.com/Azure-Samples/iot-edge-opc-publisher-nodeconfiguration) as well as an [sample application for reading diagnostic information](https://github.com/Azure-Samples/iot-edge-opc-publisher-diagnostics) are provided from OPC Publisher open-source, leveraging this interface.
+
+_Hint: The samples applications are not actively maintained and may be outdated._
+
+## Direct Methods of version 2.8.2
```
Comment (cristipogacean) [inline docs/modules/publisher-dmmigration.md:35]: I think the sections below belong in the previous doc, where direct methods are described. I mean terminology,  payload schema, and so on. I think this information is more relevant for the new users of 2.8.2, that are not interested in migration steps. 

 We could add references to this information in the migration doc,

**Comment by karok2m** [comment]
@monikavar, can you please install [markdownlint](https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint) VS Code extension and have a look at all the warnings that it is reporting ?

File: docs/modules/publisher-dmmigration.md [hunk 24]
```diff
@@ -0,0 +1,141 @@
...
+
+
+## Direct Methods of version 2.5.x
+
+The following  table describes the direct methods which were available in OPC Publisher 2.5.x with request and response. The direct methods which are removed in version 2.8.2 are `GetDiagnosticLog`, `GetDiagnosticStartupLog`, `ExitApplication` and `GetInfo`.
+
+| 2.5.x                            |                                                              |                                                              |                          |
+| -------------------------------- | ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------------ |
+| **MethodName**                   | **Request**                                                  | **Response**                                                 | **in 2.8.2** |
+| **PublishNodes**                 | EndpointUrl, List<OpcNodes>,  UseSecurity, UserName, Password | Status, List<StatusResponse>                                 | Yes                        |
```
Comment (karok2m) [inline docs/modules/publisher-dmmigration.md]: None of the lists seem to be rendered correctly, please use markdown preview functionality of VS Code to validate that they are visible. All `<` and `>` characters should have `\` before them. So instead of ```List<OpcNodes>``` you should have ```List\<OpcNodes\>```.

File: docs/modules/publisher-dmmigration.md [hunk 25]
```diff
@@ -0,0 +1,141 @@
...
+
+### Terminologies
+
+The definitions of the important terms used are described below:
+
+- **DataSet** is a group of NodeIds within an OPC UA Server to be published with the same publishing interval.
+
+- **DataSetWriter** has one DataSet and contains the elements required to successfully establish a connection to the OPC UA Server.
+
+- **DatSetGroup** is used to group several DataSetWriters for a specific OPC UA server.
```
Comment (karok2m) [inline docs/modules/publisher-dmmigration.md]: @cristipogacean, isn't `DatSetWriterGroup` the correct term instead of `DatSetGroup` ?

File: docs/modules/publisher-dmmigration.md [hunk 26]
```diff
@@ -0,0 +1,141 @@
...
+
+### Payload Schema
+
+The ` _V1` methods  uses the  payload schema as described below:
```
Comment (karok2m) [inline docs/modules/publisher-dmmigration.md]: There is space between ` and _ characters, please remove.

File: docs/modules/publisher-directmethods.md [hunk 27]
```diff
@@ -0,0 +1,328 @@
...
+- PublishNodes_V1
+- UnpublishNodes_V1
+- UnpublishAllNodes_V1
+- GetConfiguredEndpoints_V1
+- GetConfiguredNodesOnEndpoints_V1
+- GetDiagnosticInfo_V1
+
+New direct methods:
+
+-  AddOrUpdateEndpoints_V1
```
Comment (karok2m) [inline docs/modules/publisher-directmethods.md]: There are two space characters in between - and AddOrUpdateEndpoints_V1, please make it one.

File: docs/modules/publisher-directmethods.md [hunk 28]
```diff
@@ -0,0 +1,328 @@
...
+
+  *Response:* when successful - Status 200 and an empty json (`{}`) as Payload
+
+  *Exceptions:* a response corresponding to an exception will be returned if:
+  - request payload contains an endpoint (DataSet) that is not present in publisher configuration
+  - request payload contains a node that is not present in publisher configuration
+
+  *Example:*
+
+  > *Method Name:* `UnpublishNodes_V1`
```
Comment (karok2m) [inline docs/modules/publisher-directmethods.md]: There should be an empty line between `Method Name:` and `Request:`, otherwise request is put on the same line when rendered. Like this:

```
  > *Method Name:* `UnpublishNodes_V1`
  >
  > *Request:*
```

There are several occurrences like this in this file. Please fix all of them.

File: docs/modules/publisher-dmmigration.md [hunk 29]
```diff
@@ -0,0 +1,141 @@
...
+
+
+
+## Direct Methods of version 2.5.x
+
+The following  table describes the direct methods which were available in OPC Publisher 2.5.x with request and response. The direct methods which are removed in version 2.8.2 are `GetDiagnosticLog`, `GetDiagnosticStartupLog`, `ExitApplication` and `GetInfo`.
+
+| 2.5.x                            |                                                              |                                                              |                          |
+| -------------------------------- | ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------------ |
+| **MethodName**                   | **Request**                                                  | **Response**                                                 | **in 2.8.2** |
```
Comment (karok2m) [inline docs/modules/publisher-dmmigration.md:17]: `Request` column listing specific fields and not a JSON looks strange to me. I would still ask for a sample of request payload for `PublishNodes` to be added in the document, similar to what we have for `2.8.2`.

File: docs/modules/publisher-directmethods.md (hunk 22 above)
Comment (marcschier) [reply docs/modules/publisher-directmethods.md]: It should be possible to keep the same "client" working against 2.5.  I think _V1 is not required, but if we ever did a _V2 it would default to that.  Let's make sure.

File: docs/modules/publisher-dmmigration.md (hunk 25 above)
Comment (cristipogacean) [reply docs/modules/publisher-dmmigration.md]: yes, it's DatSetWriterGroup

File: docs/modules/publisher-directmethods.md (hunk 28 above)
Comment (monikavar) [reply docs/modules/publisher-directmethods.md]: I was using Typora so many things were already rendered properly for me.
Added an extra line.

File: docs/modules/publisher-dmmigration.md (hunk 23 above)
Comment (monikavar) [reply docs/modules/publisher-dmmigration.md:35]: Makes sense. Since I added request and response for 2.5.x as well in migration doc, I will move these terms in another doc.

File: docs/modules/publisher-directmethods.md (hunk 22 above)
Comment (monikavar) [reply docs/modules/publisher-directmethods.md]: I verified and _V1 is indeed required. This would actually be a change using 2.8.2 as _V1 is not required 2.5.x. I will rephrase this to be more evident.

Comment (monikavar) [reply docs/modules/publisher-directmethods.md]: Based on the discussion, _V1 should not be required. Made the changes

File: docs/modules/publisher-directmethods.md [hunk 30]
```diff
@@ -0,0 +1,427 @@
...
+  > _Method Name_: `GetConfiguredEndpoints_V1`
+  >
+  > _Response_:
+  >
+  > ```json
+  > {
+  >      "status":200,
+  >      "payload":[
+  >         {
+  >             "EndpointUrl":"opc.tcp://opcplc:50000/",
```
Comment (dacolgit) [inline docs/modules/publisher-directmethods.md:252]: Need to add endpoints object in the response. It's added here:
[https://github.com/Azure/Industrial-IoT/pull/1595/files](https://github.com/Azure/Industrial-IoT/pull/1595/files)

File: docs/modules/publisher-directmethods.md [hunk 31]
```diff
@@ -0,0 +1,427 @@
...
+  > ```
+  >
+  > _Response_:
+  >
+  > ```json
+  > {
+  >      "status":200,
+  >      "payload":[
+  >         {
+  >             "id":"nsu=http://microsoft.com/Opc/OpcPlc/;s=SlowUInt1",
```
Comment (dacolgit) [inline docs/modules/publisher-directmethods.md:293]: need to add opcNodes object in the response. it's added here:
[https://github.com/Azure/Industrial-IoT/pull/1595/files](https://github.com/Azure/Industrial-IoT/pull/1595/files)

File: docs/modules/publisher-directmethods.md [hunk 32]
```diff
@@ -0,0 +1,427 @@
...
+  >          "OutgressInputBufferCount":"0",
+  >          "OutgressInputBufferDropped":"0",
+  >          "OutgressIoTMessageCount":"0",
+  >          "ConnectionRetries":"0",
+  >          "OpcEndpointConnected":"true",
+  >          "MonitoredOpcNodesSucceededCount":"5",
+  >          "MonitoredOpcNodesFailedCount":"0"
+  >       }
+  >    ]
+  >    }
```
Comment (karok2m) [inline docs/modules/publisher-directmethods.md]: Extra closing bracket. Should be removed.

File: docs/modules/publisher-directmethods.md (hunk 30 above)
Comment (monikavar) [reply docs/modules/publisher-directmethods.md:252]: I have added a note to update this after backwards compatibility issues are fixed/merged.

File: docs/modules/publisher-directmethods.md (hunk 31 above)
Comment (monikavar) [reply docs/modules/publisher-directmethods.md:293]: I have added a note to update this after backwards compatibility issues are fixed/merged.

File: docs/modules/publisher-directmethods.md (hunk 32 above)
Comment (monikavar) [reply docs/modules/publisher-directmethods.md]: fixed

**Review by karok2m — APPROVED** [review]

**Review by cristipogacean — APPROVED** [review]
:shipit:
